core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -377,6 +402,12 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
         .map(m => s"${Utils.bytesToString(m.bytesWritten)}")
         .getOrElse("")
 
+      val maybeShuffleReadBlockedTime = metrics.flatMap(_.shuffleReadMetrics).map(_.fetchWaitTime)
+      val shuffleReadBlockedTimeSortable = maybeShuffleReadBlockedTime.map(_.toString).getOrElse("")
+      val shuffleReadBlockedTimeReadable = maybeShuffleReadBlockedTime.map{
```

Comment:
_Suuuuper_ nitpicky style nit, but I'd put a space before this `{`.  However, I think you can just write this whole thing as a one-liner:

```scala
val shuffleReadBlockedTimeReadable = 
  maybeShuffleReadBlockedTime.map(UIUtils.formatDuration).getOrElse("")
```